cmd/virt-tail/main.go

```diff
@@ -0,0 +1,181 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ *
+ */
+
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/nxadm/tail"
+	"github.com/spf13/pflag"
+	"golang.org/x/sync/errgroup"
+
+	"kubevirt.io/client-go/log"
+)
+
+var gLogFile string
+
+var gctx context.Context
+var g *errgroup.Group
+
+const initialSocketTimeout = time.Second * 20
+
+func tailLogs() error {
+	t, err := tail.TailFile(gLogFile, tail.Config{
+		Follow:        true,
+		CompleteLines: true,
+		MustExist:     false,
+		ReOpen:        true,
+		Logger:        tail.DiscardingLogger,
+	})
+	if err != nil {
+		return err
+	}
+	defer func() {
+		serr := t.Stop()
+		if serr != nil {
+			log.Log.Errorf("tail error: %v", serr)
+		}
+		t.Cleanup()
+	}()
+
+	for {
+		select {
+		case line, ok := <-t.Lines:
+			if !ok {
+				return errors.New("tail error")
+			}
+			if line.Err != nil {
+				return line.Err
+			}
+			fmt.Println(line.Text)
+		case <-gctx.Done():
+			return gctx.Err()
+		}
+	}
+}
+
+func checkFile(socketFile string, fileExists bool) bool {
+	if !fileExists {
+		_, err := os.Stat(socketFile)
+		return !os.IsNotExist(err)
+	}
+	return true
+}
+
+func watchFS() error {
+	socketFile := strings.TrimSuffix(gLogFile, "-log")
+	termFile := gLogFile + "-sigTerm"
+	socketExists := checkFile(socketFile, false)
```

Comment:
This will initialize `socketExists` always to `true`? It sounds counter intuitive to the variables name, as the socket likely does not exist yet?